cwl/src/main/scala/cwl/CwlCodecs.scala

```diff
@@ -22,5 +31,44 @@ object CwlCodecs {
   implicit val cltD = implicitly[Decoder[CommandLineTool]]
   implicit val etD = implicitly[Decoder[ExpressionTool]]
 
-  def decodeCwl(in: String): Checked[CwlFile] = decodeAccumulating[CwlFile](in).leftMap(_.map(_.getMessage).map(s"error parsing: $in" + _)).toEither
+  def errorToStrings : NonEmptyList[Error] => NonEmptyList[String] = _.map(_.show)
+
+  def decodeWithErrorStrings[A](in: String)(implicit d: Decoder[A]): ErrorOr[A] = decodeAccumulating[A](in).leftMap(errorToStrings)
+
+  /**
+    * Attempt to parse as an array of CWL or a single one.
+    *
+    * If it fails, attempt to parse each CWL individually and return a superset of those failures.
+    */
+  def decodeCwl(in: String): Checked[CwlFile] =
+    decodeWithErrorStrings[CwlFile](in).leftMap(decodePieces(in)).toEither
+
+  /**
+    * Drop down to a lower level and try to parse each CWL individually, returning a superset of those failures.
+    */
+  def decodePieces(in: String)(original: NonEmptyList[String]):NonEmptyList[String]  =
```

Comment:
I think I'm getting confused with the parameter names... isn't `in` the original CWL and `original` the error strings from the first parse?

Reply:
Yes. These will be made more accurate and descriptive.